edge/pkg/edgehub/common/http/http_test.go

```diff
@@ -105,3 +110,168 @@ func TestNewHTTPSClient(t *testing.T) {
 		})
 	}
 }
+
+//TestNewHTTPClientWithCA() tests the creation of a new HTTP using filled capem
+func TestNewHTTPClientWithCA(t *testing.T) {
+	err := util.GenerateTestCertificate("/tmp/kubeedge/testData/", "edge", "edge")
+	if err != nil {
+		t.Errorf("Error in generating fake certificates: %v", err)
+		return
+	}
+	capem, err := ioutil.ReadFile(CertFile)
+	if err != nil {
+		t.Errorf("Error in loading Cert file: %v", err)
+		return
+	}
+	certificate := tls.Certificate{}
+
+	testPool := x509.NewCertPool()
+	if ok := testPool.AppendCertsFromPEM(capem); !ok {
+		t.Errorf("cannot parse the certificates")
+		return
+	}
+
+	type args struct {
+		capem       []byte
+		certificate tls.Certificate
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *http.Client
+		wantErr bool
+	}{
+		{
+			name: "TestNewHTTPClientWithCA: ",
+			args: args{
+				capem:       capem,
+				certificate: certificate,
+			},
+			want: &http.Client{
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						RootCAs:            testPool,
+						InsecureSkipVerify: false,
+						Certificates:       []tls.Certificate{certificate},
+					},
+				},
+				Timeout: connectTimeout,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Wrong certifcate given when getting HTTP client",
+			args: args{
+				capem:       []byte{},
+				certificate: certificate,
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewHTTPClientWithCA(tt.args.capem, tt.args.certificate)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewHTTPClientWithCA() error = %v, expectedError = %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewHTTPClientWithCA() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+//TestBuildRequest() tests the process of message building
```

Comment:
```suggestion
// TestBuildRequest() tests the process of message building
```

Reply:
Thanks for the review. I hope all your comments are included in my new commit :)